lte/gateway/c/core/oai/tasks/s1ap/s1ap_state_manager.cpp

```diff
@@ -98,32 +100,23 @@ void S1apStateManager::create_state() {
   create_s1ap_imsi_map();
 }
 
-void free_s1ap_state(s1ap_state_t* state_cache_p) {
+void free_s1ap_state(oai::S1apState* state_cache_p) {
   AssertFatal(state_cache_p,
-              "s1ap_state_t passed to free_s1ap_state must not be null");
+              "S1apState passed to free_s1ap_state must not be null");
 
-  oai::EnbDescription* enb;
+  oai::EnbDescription enb;
```

Comment:
Can you move the declaration into the for loop that uses it or use itr->second without assignment?